core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -54,29 +55,29 @@ public int compare(ECKey k1, ECKey k2) {
     };
 
     private final DeterministicKey parent;
-    private final ImmutableList<ChildNumber> childNumberPath;
+    private final HDPath childNumberPath;
     private final int depth;
     private int parentFingerprint; // 0 if this key is root node of key hierarchy
 
     /** 32 bytes */
     private final byte[] chainCode;
 
     /** Constructs a key from its components. This is not normally something you should use. */
-    public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
+    public DeterministicKey(List<ChildNumber> childNumberPath,
```

Comment:
Why not migrate these constructors to HDPath too?

Reply:
Because that would be (perhaps unnecessarily) breaking. The idea is that `List<ChildNumber>` will accept both `ImmutableList<ChildNumber> ` and `HDPath`. If you'd prefer migrating those to `HDPath`, I'd be happy to change it.

Reply:
... alternatively we could use overloading to accept either with the `List` variant being deprecated. We also have to avoid the _Catch-22_ situation where we need an `HDPath` to create an `HDPath`...

Reply:
I see. Guess it's ok for now, thanks for the clarification.